flytekit/core/promise.py

```diff
@@ -94,16 +96,19 @@ def my_wf(in1: int, in2: int) -> int:
         t = native_types[k]
         try:
             if type(v) is Promise:
-                v = resolve_attr_path_in_promise(v)
-            result[k] = TypeEngine.to_literal(ctx, v, t, var.type)
+                v = await resolve_attr_path_in_promise(v)
+            result[k] = await TypeEngine.async_to_literal(ctx, v, t, var.type)
         except TypeTransformerFailedError as exc:
             exc.args = (f"Failed argument '{k}': {exc.args[0]}",)
             raise
 
     return result
 
 
-def resolve_attr_path_in_promise(p: Promise) -> Promise:
+translate_inputs_to_literals = loop_manager.synced(_translate_inputs_to_literals)
+
+
+async def resolve_attr_path_in_promise(p: Promise) -> Promise:
```

Comment:
Should we be concerned with backward compatibility with changing a function from sync to async?

Reply:
this particular function is pretty limited in terms of callees so hopefully that's okay.